sys-process/btop/files/btop-1.2.13-allow-clang.patch

```diff
@@ -0,0 +1,234 @@
++	#include <semaphore>
++#endif
+ 
+ #include <btop_shared.hpp>
+ #include <btop_tools.hpp>
+@@ -321,14 +326,12 @@ namespace Runner {
+ 	atomic<bool> coreNum_reset (false);
+ 
+ 	//* Setup semaphore for triggering thread to do work
+-#if __GNUC__ < 11
+-	#include <semaphore.h>
++#if !defined(__clang__) && __GNUC__ < 11
+ 	sem_t do_work;
+ 	inline void thread_sem_init() { sem_init(&do_work, 0, 0); }
+ 	inline void thread_wait() { sem_wait(&do_work); }
+ 	inline void thread_trigger() { sem_post(&do_work); }
+ #else
+-	#include <semaphore>
+ 	std::binary_semaphore do_work(0);
+ 	inline void thread_sem_init() { ; }
+ 	inline void thread_wait() { do_work.acquire(); }
+@@ -608,7 +611,7 @@ namespace Runner {
+ 				<< Term::sync_end << flush;
+ 		}
+ 		//* ----------------------------------------------- THREAD LOOP -----------------------------------------------
+-		pthread_exit(NULL);
++		return {};
+ 	}
+ 	//? ------------------------------------------ Secondary thread end -----------------------------------------------
+ 
+--- a/src/btop_config.cpp
++++ b/src/btop_config.cpp
+@@ -332,7 +332,7 @@ namespace Config {
+ 			new_presets.push_back(preset);
+ 		}
+ 
+-		preset_list = move(new_presets);
++		preset_list = std::move(new_presets);
+ 		return true;
+ 	}
+ 
+@@ -515,7 +515,7 @@ namespace Config {
+ 		for (auto& box : new_boxes) {
+ 			if (not v_contains(valid_boxes, box)) return false;
+ 		}
+-		current_boxes = move(new_boxes);
++		current_boxes = std::move(new_boxes);
+ 		return true;
+ 	}
+ 
+--- a/src/btop_tools.cpp
++++ b/src/btop_tools.cpp
+@@ -17,6 +17,7 @@ tab-size = 4
+ */
+ 
+ #include <cmath>
++#include <codecvt>
+ #include <iostream>
+ #include <fstream>
+ #include <ctime>
+@@ -151,7 +152,7 @@ namespace Term {
+ 
+ //? --------------------------------------------------- FUNCTIONS -----------------------------------------------------
+ 
+-// ! Dsiabled due to issue when compiling with musl, reverted back to using regex
++// ! Disabled due to issue when compiling with musl, reverted back to using regex
+ // namespace Fx {
+ // 	string uncolor(const string& s) {
+ // 		string out = s;
+--- a/src/linux/btop_collect.cpp
++++ b/src/linux/btop_collect.cpp
+@@ -1079,9 +1079,9 @@ namespace Mem {
+ 				bool new_ignored = false;
+ 				for (auto& [mountpoint, disk] : disks) {
+ 					if (std::error_code ec; not fs::exists(mountpoint, ec) or v_contains(ignore_list, mountpoint)) continue;
+-					struct statvfs64 vfs;
+-					if (statvfs64(mountpoint.c_str(), &vfs) < 0) {
+-						Logger::warning("Failed to get disk/partition stats for mount \""+ mountpoint + "\" with statvfs64 error code: " + to_string(errno) + ". Ignoring...");
++					struct statvfs vfs;
++					if (statvfs(mountpoint.c_str(), &vfs) < 0) {
++						Logger::warning("Failed to get disk/partition stats for mount \""+ mountpoint + "\" with statvfs error code: " + to_string(errno) + ". Ignoring...");
+ 						ignore_list.push_back(mountpoint);
+ 						new_ignored = true;
+ 						continue;
```

Comment:
This part of the patch doesn't really apply to the description “allow clang”. Also, I don't see a reason for all of these changes to be backported. For the sake of keeping the patch short and simple, please reduce the changes to what is necessary to make the clang build and the musl build work.

Reply:
I'll go ahead and split the patch into one for musl and one for allowing compilation with clang 16. I'm pretty sure all of the changes are needed for either clang or musl, but I'll go ahead and make sure. It was initially in one patch because it was all one commit.

Reply:
Yes, this will be needed for musl-1.2.4 (statvfs64).